### PR TITLE
Explicitely specify serialized properties

### PR DESCRIPTION
**Pull request by davidbgk**
Introducing the `keys` options to reduce the generated file to chosen properties and have the ability not to export geometries.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -179,7 +182,7 @@ Perform some non geospatial processing (ex: set the postal codes, attach the par
 ``dist``
 ~~~~~~~~
 
-Dump the produced dataset as GeoJSON files for distribution.
+Dump the produced dataset as GeoJSON or MSGPack files for distribution.
```
Comment (jdesboeufs) [inline README.rst]: How do we choose GeoJSON or MSGPack distributions?

Comment (davidbgk) [inline README.rst]: Using the `-r/--serialization` option. I'll add missing documentation for that option too.

File: README.rst [hunk 2]
```diff
@@ -227,6 +230,19 @@ many gigabytes of RAM given that it can iterate over data without loading
 the whole file.
 
 
+``keys``
+~~~~~~~~
+
+You can export only the properties that you want to use.
+This option only works with the JSON serialization.
```
Comment (jdesboeufs) [inline README.rst]: Why not prefer something like this?
```bash
./geozones.py dump output.json # For GeoJSON/default output
./geozones.py dump | json2msgpackkillerclitool > somewhere.msgpack
```

There is another formats, like [geobuf](https://github.com/mapbox/geobuf), used by some parts of JS ecosystem.
```bash
./geozones.py dump | json2geobuf > output.pbf
```

File: README.rst [hunk 3]
```diff
@@ -227,6 +230,19 @@ many gigabytes of RAM given that it can iterate over data without loading
...
+
+.. code-block:: shell
+
+    # Only export population and area without geometry
+    $ ./geozones.py dist -k population,area
```
Comment (jdesboeufs) [inline README.rst]: ❤️

File: geozones.py [hunk 4]
```diff
@@ -147,10 +147,12 @@ def postprocess(ctx, only):
 @click.option('-c/-nc', '--compress/--no-compress', default=True)
 @click.option('-r', '--serialization', default='json',
               type=click.Choice(['json', 'msgpack']))
-def dist(ctx, pretty, split, compress, serialization):
+@click.option('-k', '--keys', default=None)
+def dist(ctx, pretty, split, compress, serialization, keys):
     '''Dump a distributable file'''
-    title('Dumping data to {serialization}'.format(
-        serialization=serialization))
+    keys = keys and keys.split(',')
```
Comment (noirbizarre) [inline geozones.py]: I think you can let click handle the multi arguments: http://click.pocoo.org/5/options/#multi-value-options

**Comment by noirbizarre** [comment]
r+wc 

Nice !

File: geozones.py (hunk 4 above)
Comment (davidbgk) [inline geozones.py]: I read the whole documentation and click needs the number of parameters with multi value :(

File: README.rst (hunk 2 above)
Comment (davidbgk) [inline README.rst]: I found https://github.com/tv42/msgpack-json in go (not updated for 3 years :/)

File: geojson.py [hunk 5]
```diff
@@ -4,31 +4,38 @@
...
-            'population': zone.get('population', None),
-            'area': zone.get('area', None),
-            'flag': unicodify(zone.get('flag', '')) or None,
-            'blazon': unicodify(zone.get('blazon', '')) or None,
-            'keys': zone.get('keys', {}),
-            'parents': zone.get('parents', [])
-        }
+        'properties': properties
     }
+    if keys is not None and 'geometry' not in keys:
```
Comment (jdesboeufs) [inline geojson.py]: So GeoJSON is not valid.
You should prefer a regular JSON output in this case, with properties in root objects.

File: README.rst (hunk 2 above)
Comment (davidbgk) [inline README.rst]: I need to think about it. The current approach is to pass options to generate the pertinent dataset but we can think of it as a stream reduced afterwards. It requires to generate the whole file first.

File: geojson.py (hunk 5 above)
Comment (davidbgk) [inline geojson.py]: Oh, good point.

**Pull request merged**